File: file_writer_output/y_combinator.js
// Y Combinator function in JavaScript

const Y = f => (x => x(x))(x => f(y => x(x)(y)));

// Example usage:
const factorial = Y(fac => n => n <= 1 ? 1 : n * fac(n - 1));

console.log(factorial(5));  // Output: 120
console.log(factorial(10)); // Output: 3628800

File: file_writer_output/ycombinator.nim
# Y Combinator implementation in Nim

proc Y[T, R](f: proc(f: proc(x: T): R): proc(x: T): R): proc(x: T): R =
  proc g(x: T): R =
    f(Y(f))(x)
  return g

# Example usage:
proc factorial(f: proc(n: int): int): proc(n: int): int =
  return proc(n: int): int =
    if n <= 1: 1
    else: n * f(n - 1)

let fact = Y(factorial)
echo fact(5)  # Output: 120
echo fact(10) # Output: 3628800
